src/handshakes/abstract-handshake.ts

```diff
@@ -106,7 +106,8 @@ export abstract class AbstractHandshake {
 
   protected dh(privateKey: bytes32, publicKey: bytes32): bytes32 {
     try {
-      return Buffer.from(box.before(publicKey, privateKey), 0, 32);
+      const sharedKey = box.before(publicKey, privateKey)
+      return Buffer.from(sharedKey);
```

Comment:
```suggestion
      return Buffer.from(sharedKey.buffer, sharedKey.offset, sharedKey.length);
```
It won't allocate new array